transport/src/main/java/io/zeebe/transport/impl/sender/Sender.java

```diff
@@ -198,8 +188,13 @@ private void onRequestSubmitted(final OutgoingRequest request) {
           actor.runDelayed(Duration.ofMillis(10), () -> submittedRequests.offer(request));
         }
       } else {
-        // no remote address available, retry
-        actor.runDelayed(Duration.ofMillis(10), () -> submittedRequests.offer(request));
+        if (request.shouldRetry(
+            new IncomingResponse(
```

Comment:
Can we declare this as a variable instead of passing it as is, so we can name it in a way that it makes it explicit we're generating a failed response to propagate that no remote address was found? e.g. `final IncomingResponse failedResponse = ...` or something.